lib/helperFunctions.js

```diff
@@ -1,7 +1,118 @@
+import axios from "axios";
+import { aboutMe, itemsToFetch, includedRepos } from "../src/constants";
+
 export const scrollToSection = (id) => {
   const element = document.getElementById(id);
   const yOffset = -70; // Adjust this value based on your header height
   const y = element.getBoundingClientRect().top + window.scrollY + yOffset;
 
   window.scrollTo({ top: y, behavior: "smooth" });
 };
+
+const parseOriginFromUrl = (url) => {
+  /**
+   * splits https://github.com/repos/org-name/repo-name/issues/25
+   * into [ "github.com", "repos", "org-name", "repo-name", "issues", "25"]
+   */
+  const [, ...parts] = url.split(/https:\/\/|\//gm);
+  const organization = parts[1];
+  const repo = parts[2];
+  /**
+   * accessing a github profile or organization and adding a .png
+   * at the end of the URL will return their logo/profile picture
+   */
+  const logoUrl = `https://github.com/${organization}.png`;
+
+  return {
+    organization,
+    repo,
+    logoUrl,
+  };
+};
+
+export async function fetchContributionsWithRetry(maxRetries = 1) {
+  let attempts = 0;
+
+  while (attempts <= maxRetries) {
+    try {
+      const result = await fetchContributions();
+      return result; // If successful, return the result
+    } catch (error) {
+      attempts++;
+      console.log(`Attempt ${attempts} failed: ${error.message}. Retrying...`);
+
+      if (attempts > maxRetries) {
+        console.log("Max retries reached. Returning last error.");
+        // Optionally, return a formatted error message or throw
+        return { error: error.message };
+      }
+    }
+  }
+}
+
+
+function generatePRQuery(repos, username) {
+  const queries = repos
+    .map((repo) => {
+      return `repo:${repo} is:pr author:${username}`;
+    })
+    .join(" ");
+
+  return `
+    query {
+      search(query: "${queries}", type: ISSUE, first: 12) {
```

Comment:
Lets not hardcode 12 here, use itemsToFetch

Reply:
edited query to use constant _itemsToFetch_